packages/@react-spectrum/cards/src/CardBase.tsx

```diff
@@ -0,0 +1,190 @@
+import {useProviderProps} from '@react-spectrum/provider';
+
+interface CardBaseProps<T> extends SpectrumCardProps {
+  articleProps?: HTMLAttributes<HTMLElement>,
+  item?: Node<T>
+}
+
+function CardBase<T extends object>(props: CardBaseProps<T>, ref: DOMRef<HTMLDivElement>) {
+  props = useProviderProps(props);
+  // TODO: Don't send in articleProps via context (unless we want to make another context for InternalCard)? Pass it in via props since it will only be provided via CardView's InternalCard
+  let context = useCardViewContext() || {}; // we can call again here, won't change from Card.tsx
+  let {state} = context;
+  let manager = state?.selectionManager;
+  // let [isSelected, setIsSelected] = useControlledState(context.isSelected, undefined, context.onSelectionChange);
+  let {
+    isQuiet,
+    orientation = 'vertical',
+    articleProps = {},
+    item,
+    layout
+  } = props;
+
+  let key = item?.key;
+  let isSelected = manager?.isSelected(key);
+  let isDisabled = state?.disabledKeys.has(key);
+  let onChange = () => manager.select(key);
+
+  let {styleProps} = useStyleProps(props);
+  let {cardProps, titleProps, contentProps} = useCard(props);
+  let domRef = useDOMRef(ref);
+  let gridRef = useRef<HTMLDivElement>();
+
+  let {hoverProps, isHovered} = useHover({});
+  let [isFocused, setIsFocused] = useState(false);
+  let {focusWithinProps} = useFocusWithin({
+    onFocusWithinChange: setIsFocused,
+    isDisabled
+  });
+
+  let hasFooter = useHasChild(`.${styles['spectrum-Card-footer']}`, gridRef);
+
+  let slots = useMemo(() => ({
+    image: {UNSAFE_className: classNames(styles, 'spectrum-Card-image'), objectFit: orientation === 'horizontal' ? 'cover' : 'contain', alt: ''},
+    illustration: {UNSAFE_className: classNames(styles, 'spectrum-Card-illustration')},
+    avatar: {UNSAFE_className: classNames(styles, 'spectrum-Card-avatar'), size: 'avatar-size-100'},
+    heading: {UNSAFE_className: classNames(styles, 'spectrum-Card-heading'), ...titleProps},
+    content: {UNSAFE_className: classNames(styles, 'spectrum-Card-content'), ...contentProps},
+    detail: {UNSAFE_className: classNames(styles, 'spectrum-Card-detail')},
+    actionmenu: {UNSAFE_className: classNames(styles, 'spectrum-Card-actions'), align: 'end', isQuiet: true},
+    footer: {UNSAFE_className: classNames(styles, 'spectrum-Card-footer'), isHidden: isQuiet},
+    divider: {UNSAFE_className: classNames(styles, 'spectrum-Card-divider'), size: 'S'}
+  }), [titleProps, contentProps]);
+
+  // this is for horizontal cards
+  let [height, setHeight] = useState(NaN);
+  useLayoutEffect(() => {
+    if (orientation !== 'horizontal') {
+      return;
+    }
+    let cardHeight = gridRef.current.getBoundingClientRect().height;
+    setHeight(cardHeight);
+  }, [gridRef, setHeight]);
+
+  // This is only for quiet grid cards
+  let [isCloseToSquare, setIsCloseToSquare] = useState(false);
+  useLayoutEffect(() => {
```

Comment:
yeah, I do not know, but i do worry about it, we could memo some stuff if it becomes an issue, though it might make it harder to know when to update it
another option is that we could push it up into cardview and just have a prop on cards that says 'nearly square' or something
cardview has all of the information already, the ratio of the image and it calculates the target dimensions that card will fit into whereas card has no idea until it is rendered what any of this information is

Reply:
We've seen the start of this in Daniel's PR, so far, no performance issues seen.